openmc/deplete/results.py

```diff
@@ -1,521 +1,417 @@
 
+        Returns
+        -------
+        mat_file : Materials
+            A modified Materials instance containing depleted material data
+            and original isotopic compositions of non-depletable materials
         """
-
-        if not model.materials:
-            materials = openmc.Materials(
-                model.geometry.get_all_materials().values()
-            )
+        result = self[burnup_index]
+
+        # Only materials found in the original materials.xml file will be
+        # updated. If for some reason you have modified OpenMC to produce
+        # new materials as depletion takes place, this method will not
+        # work as expected and leave out that material.
+        mat_file = Materials.from_xml("materials.xml")
+
+        # Only nuclides with valid transport data will be written to
+        # the new materials XML file. The precedence of nuclides to select
+        # is first ones provided as a kwarg here, then ones specified
+        # in the materials.xml file if provided, then finally from
+        # the environment variable OPENMC_CROSS_SECTIONS.
+        if nuc_with_data:
+            cv.check_iterable_type('nuclide names', nuc_with_data, str)
+            available_cross_sections = nuc_with_data
         else:
-            materials = model.materials
-
-        for material in materials:
-            if material.depletable:
-                material.volume = self.volume[str(material.id)]
+            # select cross_sections.xml file to use
+            if mat_file.cross_sections:
+                this_library = DataLibrary.from_xml(path=mat_file.cross_sections)
+            else:
+                this_library = DataLibrary.from_xml()
+
+            # Find neutron libraries we have access to
+            available_cross_sections = set()
+            for lib in this_library.libraries:
+                if lib['type'] == 'neutron':
+                    available_cross_sections.update(lib['materials'])
+            if not available_cross_sections:
+                raise DataError('No neutron libraries found in cross_sections.xml')
+
+        # Overwrite material definitions, if they can be found in the depletion
+        # results, and save them to the new depleted xml file.
+        for mat in mat_file:
+            mat_id = str(mat.id)
+            if mat_id in result.mat_to_ind:
+                mat.volume = result.volume[mat_id]
+                mat.set_density('sum')
+                for nuc in result.nuc_to_ind:
+                    if nuc not in available_cross_sections:
+                        continue
+                    atoms = result[0, mat_id, nuc]
+                    if atoms > 0.0:
+                        atoms_per_barn_cm = 1e-24 * atoms / mat.volume
+                        mat.remove_nuclide(nuc) # Replace if it's there
+                        mat.add_nuclide(nuc, atoms_per_barn_cm)
+
+        return mat_file
+
+
+# Retain deprecated name for the time being
+ResultsList = Results
```

Comment:
Previous API is to do `ResultsList.from_hdf5` so there likely isn't a large user base who would miss any warnings going from `ResultsList(...)` to `Results(...)`

More thinking if it would be beneficial to provide `ResultsList` as a subclass with a warning in the `__init__` method. But since the recommended way before was to do `ResultsList.from_hdf5` I think I've talked myself out of this suggestion :sweat_smile:

Reply:
Yeah, the main goal here is that if someone calls `ResultsList.from_hdf5`, they are given a warning telling them the new way to do it.